actions/user.ts

```diff
@@ -0,0 +1,67 @@
+'use server';
+
+import prisma from '@/prisma';
+import { type User } from '@prisma/client';
+import { del, put } from '@vercel/blob';
+import { revalidatePath } from 'next/cache';
+
+const updateUserProfile = async (user_id: string, user: User) => {
+  try {
+    const new_user = await prisma.user.update({
+      where: {
+        id: user_id,
+      },
+      data: { ...user },
+    });
+
+    return new_user;
+  } catch {
+    return null;
+  }
+};
+
+const updateUserAvatar = async (user_id: string, formData: FormData) => {
+  const avatar_image = formData.get('file') as File;
+
+  const blob = await put(avatar_image.name, avatar_image, {
+    access: 'public',
+  });
+
+  revalidatePath('/');
+
```

Comment:
You should delete previous avatar before before uploading and setting new one here! That way we don't forget to call "`delete`".